File: veriq-frontend/src/services/departmentService.ts
import { apiClient } from './api/apiClient';
import { ApiResponse } from './api/apiTypes';

export interface DepartmentDTO {
  id: string;
  code: string;
  name: string;
  organizationId?: string;
  status: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface CreateDepartmentPayloadDTO {
  code: string;
  name: string;
  status?: string;
}

export const departmentService = {
  getAllDepartments: async (): Promise<DepartmentDTO[]> => {
    const res = await apiClient.get<ApiResponse<DepartmentDTO[]>>('/departments');
    return res.data.data || [];
  },

  getDepartmentById: async (id: string): Promise<DepartmentDTO> => {
    const res = await apiClient.get<ApiResponse<DepartmentDTO>>(`/departments/${id}`);
    return res.data.data;
  },

  createDepartment: async (payload: CreateDepartmentPayloadDTO): Promise<DepartmentDTO> => {
    const res = await apiClient.post<ApiResponse<DepartmentDTO>>('/departments', {
      name: payload.name,
      code: payload.code,
      status: payload.status || 'ACTIVE'
    });
    return res.data.data;
  },

  updateDepartment: async (id: string, payload: Partial<CreateDepartmentPayloadDTO>): Promise<DepartmentDTO> => {
    const res = await apiClient.put<ApiResponse<DepartmentDTO>>(`/departments/${id}`, payload);
    return res.data.data;
  },

  deleteDepartment: async (id: string): Promise<void> => {
    await apiClient.delete<ApiResponse<void>>(`/departments/${id}`);
  }
};

File: veriq-backend/src/main/java/com/veriq/department/service/DepartmentService.java
package com.veriq.department.service;

import com.veriq.department.dto.CreateDepartmentPayloadDTO;
import com.veriq.department.dto.DepartmentDTO;
import com.veriq.department.dto.UpdateDepartmentPayloadDTO;

import java.util.List;
import java.util.UUID;

public interface DepartmentService {

    List<DepartmentDTO> getAllDepartments();

    DepartmentDTO getDepartmentById(UUID id);

    DepartmentDTO createDepartment(CreateDepartmentPayloadDTO payload);

    DepartmentDTO updateDepartment(UUID id, UpdateDepartmentPayloadDTO payload);

    void deleteDepartment(UUID id);
}
